README.md

```diff
@@ -1,30 +1,55 @@
+
+In the Package Manager window, press the "+"-button, and then "Add package from git URL...". You can use both HTTPS and SSH to import the package.
+
+![Adding from git](https://github.com/erikbhan/UniTac/assets/42799026/fca9382e-c42a-47cb-a244-1ee49729b7ab)
 
 ## Usage
 
-After importing our package, you first need a manager to administrate the sensor. Click and drag our pre-made prefab into either the Unity Scene or the Unity Hierarchy from the Package folder. To troubleshoot the MQTT client or server, you can change the logging level in the manager prefab's Inspector window. You can also set a custom port if you'd like.
+After importing our package, you first need a manager to administrate the sensor. Click and drag our pre-made prefab into either the Unity Scene or the Unity Hierarchy from the Package folder. To troubleshoot the MQTT client or server, you can change the logging level in the manager prefab's Inspector window. You can also set a custom port if you'd like. Keep logging disabled in final product to maximize efficiency.
+
+![Manager inspector window](https://github.com/erikbhan/UniTac/assets/42799026/cb563644-6f36-448c-9acc-9b1c875bf64b)
+
+To handle incoming messages any sensors needs to be added. Use the "Add Sensor"-button on the manager Inspector window to add any number of sensors.
+
+![Add sensor button](https://github.com/erikbhan/UniTac/assets/42799026/2e2a6da1-d3ec-4657-a13f-740ea732c262)
+
+Configure each sensor via the Inspector window; you need to set the "serial" field equal to the physical sensor's serial number. Seconds until idle is simply the time from the last message until the sensor considers itself "idle".
 
-To handle incoming messages any sensors needs to be added. Use the "Add Sensor"-button on the manager Inspector window to add any number of sensors. Configure each sensor via the Inspector window; you need to set the "serial" field equal to the physical sensor's serial number. Seconds until idle is simply the time from the last message until the sensor considers itself "idle".
+![Sensor inspector window](https://github.com/erikbhan/UniTac/assets/42799026/cd2ff6c6-67d4-4ab5-bce4-b0b6151444d7)
 
-Finally, you need to add functionality to the sensor, either by adding a sample or your own script. Simply drag the wanted functionality onto the sensor GameObject in the Unity Hierarchy, then configure via the Inspector again (for example, if you need to make a 3D object to use as a prefab for spawning).
+Finally, you need to add functionality to the sensor, either by adding a sample or your own script. Simply drag the wanted functionality onto the sensor GameObject in the Unity Hierarchy or use the "Add component"-button in the sensor inspector window.
+
+### Sensor Events
+
+The sensor implements events to facilitate observer pattern scripts. The sensor has a low update rate so it's recommended to utilize events to update only when new data is available.
```

Comment:
```suggestion
The sensor implements the [observer pattern](https://en.wikipedia.org/wiki/Observer_pattern), utilizing [UnityEvent](https://docs.unity3d.com/ScriptReference/Events.UnityEvent.html). Because the sensor only sends a new packet every 100ms, we avoid some overhead by not checking the sensor every frame. In addition, you can add as many listeners as you want without creating unnecessary coupling in your code. 
```
mulig dette blir for lang tekst, og unødvendig, men jeg syns det burde stå ett eller annet sted